src/Selenium2Library/keywords/_screenshot.py

```diff
@@ -1,14 +1,35 @@
-import os, errno
 import robot
+import os, errno
+
+from Selenium2Library import utils
 from keywordgroup import KeywordGroup
+from robot.libraries.BuiltIn import BuiltIn, RobotNotRunningError
 
 class _ScreenshotKeywords(KeywordGroup):
 
     def __init__(self):
         self._screenshot_index = 0
+        self._screenshot_path_stack = []
```

Comment:
It's likely that this could cause an issue. If the same scope sets two screenshot paths then only the first would be popped off when the scope exited. That could cause an issue where the wrong path is used for other scopes...